docs/sql-migration-guide-upgrade.md

```diff
@@ -151,6 +151,8 @@ license: |
 
   - Since Spark 3.0, substitution order of nested WITH clauses is changed and an inner CTE definition takes precedence over an outer. In version 2.4 and earlier, `WITH t AS (SELECT 1), t2 AS (WITH t AS (SELECT 2) SELECT * FROM t) SELECT * FROM t2` returns `1` while in version 3.0 it returns `2`. The previous behaviour can be restored by setting `spark.sql.legacy.ctePrecedence.enabled` to `true`.
 
+  - Since Spark 3.0, the `add_months` function adjusts the resulting date to a last day of month only if it is invalid. For example, `select add_months(DATE'2019-01-31', 1)` results `2019-02-28`. In Spark version 2.4 and earlier, the resulting date is adjusted when it is invalid, or the original date is a last day of months. For example, adding a month to `2019-02-28` resultes in `2019-03-31`.
```

Comment:
Hm .. shall we update this migration guide? Actually `select add_months(DATE'2019-01-31', 1)` returns `2019-02-28` in both the current master and the versions before Spark 2.4.x.

I think we should explicitly mention `select add_months(DATE'2019-02-28', 1)` case only:

```
scala> sql("select add_months(DATE'2019-02-28', 1)").show()
+--------------------------------+
|add_months(DATE '2019-02-28', 1)|
+--------------------------------+
|                      2019-03-31|
+--------------------------------+
```

```
scala> sql("select add_months(DATE'2019-02-28', 1)").show()
+--------------------------------+
|add_months(DATE '2019-02-28', 1)|
+--------------------------------+
|                      2019-03-28|
+--------------------------------+
```

Reply:
Let me update it soon.